cli/src/pcluster/resources/head_node/user_data.sh

```diff
@@ -55,7 +55,8 @@ function error_exit
 {
   # wait logs flush before signaling the failure
   sleep 10
-  cfn-signal --exit-code=1 --reason="$1" "${!wait_condition_handle_presigned_url}"
+  reason=$(cat /var/log/parallelcluster/cfn_reason_write_chef_error.txt) || reason="$1"
```

Comment:
what about calling the file `chef_error_msg` without extension?

Reply:
please update base on the new file name (ref https://github.com/aws/aws-parallelcluster-cookbook/pull/1329)

Reply:
to avoid "No such file or directory" error in the log when `/var/log/parallelcluster/cfn_reason_write_chef_error.txt` isn't present, please redirect stderr to /dev/null, e.g.

```
$(cat /var/log/parallelcluster/cfn_reason_write_chef_error.txt 2>/dev/null)
```

Reply:
Changed